io/shared/src/test/scala/fs2/io/net/tcp/SocketSuite.scala

```diff
@@ -216,5 +217,51 @@ class SocketSuite extends Fs2Suite with SocketSuitePlatform {
           assertEquals(msgs, Vector(msg))
         }
     }
+
+    test("read after timed out read not allowed on JVM") {
+      val setup = for {
+        serverSetup <- Network[IO].serverResource(Some(ip"127.0.0.1"))
+        (bindAddress, server) = serverSetup
+        client <- Network[IO].client(bindAddress)
+      } yield (server, client)
+      Stream
+        .resource(setup)
+        .flatMap { case (server, client) =>
+          val echoServer = server.flatMap(c => c.writes(c.reads).attempt)
+          val msg = Chunk.array("Hello!".getBytes)
+          val prg =
+            client.write(msg) *>
+              client.readN(msg.size) *>
+              client.readN(msg.size).timeout(100.millis).recover { case _: TimeoutException =>
+                Chunk.empty
+              } *>
+              client.write(msg) *>
+              client
+                .readN(msg.size)
+                .flatMap { c =>
+                  if (isJVM) {
+                    assertEquals(c.size, 0)
+                    // Read again now that the pending read is no longer pending
```

Comment:
I think this comment is no longer relevant since the channel is dead.

Reply:
Still accurate -- it's testing that a subsequent `readN` returns 0 size chunk instead of actually performing a read or otherwise returning previously read data.

Reply:
Ah, gotcha.